File: leetcode/412-Fizz-Buzz/answer.js
/**
 * @param {number} n
 * @return {string[]}
 */

// https://leetcode.com/problems/fizz-buzz/

var fizzBuzz = function(n) {
    
    var ret_list = [];
    for (var i = 1; i <= n; ++i) {
        var res = "";
        if (i % 3 === 0) {
            res += "Fizz";
        }
        if (i % 5 === 0) {
            res += "Buzz";
        }
        if (res === "") {
            ret_list.push(String(i));
        } else {
            ret_list.push(res);
        }
    }
    return ret_list;
};

File: leetcode/412-Fizz-Buzz/answer.c
// https://leetcode.com/problems/number-of-1-bits/

#include <stdio.h>
#include <stdlib.h>

#define MAX_LEN 16

char** fizzBuzz(int n, int* returnSize) {
    
    char **res;
    res = (char **)malloc(sizeof(char**) * n);

    int i, num;
    for (i = 0; i < n; i++) {
        res[i] = (char*)malloc(sizeof(char*) * MAX_LEN);
        num = i + 1;
        if (num % 3 == 0 && num % 5 ==0) {
            res[i] = "FizzBuzz";
        } else if (num % 3 == 0) {
            res[i] = "Fizz";
        } else if (num % 5 == 0) {
            res[i] = "Buzz";
        } else {
            sprintf(res[i], "%d", num);
        }
    }

    *returnSize = n;

    return res;
}
